httpx/response.go

```diff
@@ -23,24 +14,15 @@ type Response struct {
 	StatusCode int
 }
 
-// NewResponseOK creates a new Response for a payload
-// with HTTP 200 as the status code.
-func NewResponseOK(payload interface{}) *Response {
-	return &Response{
-		Payload:    payload,
-		StatusCode: http.StatusOK,
+// GetStatusCode returns the status code for the response.
+func (r Response) GetStatusCode() int {
+	if r.StatusCode <= 0 {
+		return http.StatusOK
 	}
+	return r.StatusCode
 }
 
-// Write writes a response to an http.ResponseWriter.
-// Any errors encountered will be automatically handled.
-func (r *Response) Write(ctx context.Context, w http.ResponseWriter) {
```

Comment:
I'm kind of a fan of this way

Reply:
I liked it initially as well but it felt too OOP and the API was awkward. This also was awkward because there was a lot of duplication between the two types so it made more sense to just have a function that does the writing.